pkg/envoy/cds/cluster.go

```diff
@@ -97,21 +96,25 @@ func getLocalServiceCluster(catalog catalog.MeshCataloger, proxyServiceName serv
 		Http2ProtocolOptions: &xds_core.Http2ProtocolOptions{},
 	}
 
-	endpoints, err := catalog.ListEndpointsForService(proxyServiceName)
+	ports, err := catalog.GetTargetPortToProtocolMappingForService(proxyServiceName)
 	if err != nil {
-		log.Error().Err(err).Msgf("Failed to get endpoints for service %s", proxyServiceName)
-		return nil, err
+		if proxyServiceName.SyntheticTargetPort != 0 {
+			ports = map[uint32]string{proxyServiceName.SyntheticTargetPort: ""}
+		} else {
+			log.Error().Err(err).Msgf("Failed to get ports for service %s", proxyServiceName)
+			return nil, err
+		}
```

Comment:
Instead of handling synthetic services, I don't see a reason why we need to create local clusters for services that do not really exist. 

Can we just check if the service name matches the syntax of a synthetic service and skip creating local clusters instead?

Reply:
If we skip creating a local cluster, the tcp e2e fails where there is no service for the client

Reply:
Ah, the classic RDS issue which #2064 is blocked on.